server/src/main/java/org/elasticsearch/tasks/TaskInfo.java

```diff
@@ -196,7 +196,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         }
         builder.field("running_time_in_nanos", runningTimeNanos);
         builder.field("cancellable", cancellable);
-        if (parentTaskId.isSet()) {
+        if (parentTaskId.isSet() && this.getAction().contains("reindex") == false) {
```

Comment:
What's the problem with setting this in case of reindex? Is it because it's null in that case?

Reply:
The problem here is that many tests depend on reindexing storing the result of the reindex task in the `.tasks` index. The `.tasks` index mapping does not allow a parent task-id. Under this change, the reindex task has a parent task (the persistent task). So I blocked the serialization to aid with development. 

We need a solution to this issue in order to maintain compatibility with many, many tests.